### PR TITLE
cargo-lambda: 0.18.1 -> 0.21.1

### PR DESCRIPTION
**Pull request by taylor1791**
## Description of changes

This updates cargo-lambda to the latest version. A changelog is available, [here](https://github.com/cargo-lambda/cargo-lambda/releases).

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [x] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [X] For nonLinux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested with [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/rust/cargo-lambda/default.nix
Comment (cafkafk) [inline pkgs/development/tools/rust/cargo-lambda/default.nix:23]: non-blocking
```suggestion
    hash = "sha256-QlTAYfd0taXfK370nzqictwK7bZ4bnh1oPBJKZzhnMo=";
```

**Comment by cafkafk** [comment]
Result of `nixpkgs-review pr 259528` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cargo-lambda</li>
  </ul>
</details>

**Review by calavera — APPROVED** [review]

**Review by calavera — APPROVED** [review]

**Review by calavera — APPROVED** [review]

**Comment by matthiasdebernardini** [comment]
what can I do to help get this merged? thank you all for your work!

**Comment by taylor1791** [comment]
@matthiasdebernardini We are waiting on someone with write access to approve and merge. I posted on the nixos discourse. That usually keeps it moving along.

**Pull request merged**
